src/Adyen/Service/AbstractPlatformsResource.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Adyen\Service;
+
+use Adyen\PlatformsService;
+
+class AbstractPlatformsResource extends AbstractResource
+{
+    protected $baseUrl;
+    protected $allowApplicationInfo = false;
+    protected $allowApplicationInfoPOS = false;
+
+    public function __construct(PlatformsService $service)
+    {
+        $this->baseUrl = $service->getBaseUrl();
+
+        parent::__construct(
+            $service,
+            $this->baseUrl
+        );
+    }
+
+    /**
+     * @param string $version
+     * @param string $endpoint
+     * @param callable $callback
+     * @return mixed
+     */
+    public function simulateEndpoint(string $version, string $endpoint, callable $callback)
```

Comment:
Hi @julian-van-drielen! I think this method adds a really interesting abstraction that could be used elsewhere in the library as well. We will discuss this approach internally before making a decision about adding this. However I feel like the name `simulateEndpoint` makes it sound like a test method (threw me off at first) when it is actually use to build the endpoint for all the requests you added. We would consider renaming it, but your ideas are welcome as well.